### ISSUE TITLE
Rule S2931 - support Disposable ref structs and using declarations

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2999]
### Description

add tests and update implementation

part of https://jira.sonarsource.com/browse/MMF-1842

**Comment by christophe-zurn-sonarsource** [comment]
@andrei-epure-sonarsource I'm not so sure it actually makes sense to extends this rule to disposable ref structs. Maybe we can discuss about this during next standup

**Comment by andrei-epure-sonarsource** [comment]
@christophe-zurn-sonarsource  I think on this one, we decided it would deserve a separate rule?

**Comment by christophe-zurn-sonarsource** [comment]
@andrei-epure-sonarsource Yes indeed. That's why I put it in `rule specification work` milestone, as the RSPEC needs to be created first (even if it should be really similar to the one for classes)

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1962](https://sonarsource.atlassian.net/browse/NET-1962)

[NET-1962]: https://sonarsource.atlassian.net/browse/NET-1962?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
